libbeat/publisher/queue/memqueue/broker.go

```diff
@@ -255,43 +261,43 @@ func (b *broker) Metrics() (queue.Metrics, error) {
 
 	return queue.Metrics{
 		EventCount:            opt.UintWith(uint64(resp.currentQueueSize)),
-		EventLimit:            opt.UintWith(uint64(b.bufSize)),
+		EventLimit:            opt.UintWith(uint64(len(b.buf))),
 		UnackedConsumedEvents: opt.UintWith(uint64(resp.occupiedRead)),
 		OldestEntryID:         resp.oldestEntryID,
 	}, nil
 }
 
-var ackChanPool = sync.Pool{
+var batchPool = sync.Pool{
 	New: func() interface{} {
-		return &batchACKState{
+		return &batch{
 			doneChan: make(chan batchDoneMsg, 1),
 		}
 	},
 }
 
-func newBatchACKState(start, count int, entries []queueEntry) *batchACKState {
-	ch := ackChanPool.Get().(*batchACKState)
-	ch.next = nil
-	ch.start = start
-	ch.count = count
-	ch.entries = entries
-	return ch
+func newBatch(queue *broker, start, count int) *batch {
+	batch := batchPool.Get().(*batch)
+	batch.next = nil
+	batch.queue = queue
+	batch.start = start
+	batch.count = count
+	return batch
 }
 
-func releaseACKChan(c *batchACKState) {
-	c.next = nil
-	ackChanPool.Put(c)
+func releaseBatch(b *batch) {
+	b.next = nil
+	batchPool.Put(b)
 }
```

Comment:
This was preexisting code so I'm not sure the original context, but I left it thinking it might be a small cleanup for the garbage collector. I also don't see any logical reason it's necessary. But the other fields aren't cleared because they don't reference any other memory (and in particular the channel is reused across multiple batches via the shared pool).

Reply:
Thanks for the clarification.